src/main/kotlin/no/nav/syfo/dialogmotekandidat/api/HistorikkDTO.kt

```diff
@@ -0,0 +1,72 @@
+package no.nav.syfo.dialogmotekandidat.api
+
+import no.nav.syfo.dialogmotekandidat.domain.DialogmotekandidatEndring
+import no.nav.syfo.dialogmotekandidat.domain.DialogmotekandidatEndringArsak
+import no.nav.syfo.ikkeaktuell.domain.IkkeAktuell
+import no.nav.syfo.unntak.domain.Unntak
+import java.time.LocalDateTime
+
+data class HistorikkDTO(
+    val tidspunkt: LocalDateTime,
+    val type: HistorikkType,
+    val arsak: String,
+    val vurdertAv: String?,
+) {
+    companion object {
+        fun createHistorikkDTOs(
+            dialogmotekandidatEndringer: List<DialogmotekandidatEndring>,
+            unntak: List<Unntak>,
+            ikkeAktuell: List<IkkeAktuell>,
+        ): List<HistorikkDTO> {
+            val historikk = dialogmotekandidatEndringer.toKandidatHistorikk() + unntak.toUnntakHistorikk() + ikkeAktuell.toIkkeAktuellHistorikk()
+            return historikk.sortedByDescending { it.tidspunkt }
+        }
+    }
+}
+
+fun List<DialogmotekandidatEndring>.toKandidatHistorikk(): List<HistorikkDTO> = this.mapNotNull {
+    val tidspunkt = it.createdAt.toLocalDateTime()
+    when (it.arsak) {
+        DialogmotekandidatEndringArsak.STOPPUNKT -> HistorikkDTO(
+            tidspunkt = tidspunkt,
+            type = HistorikkType.KANDIDAT,
+            arsak = it.arsak.name,
+            vurdertAv = null,
+        )
+
+        DialogmotekandidatEndringArsak.LUKKET -> HistorikkDTO(
+            tidspunkt = tidspunkt,
+            type = HistorikkType.LUKKET,
+            arsak = it.arsak.name,
+            vurdertAv = null,
+        )
+
+        // Disse dekkes av dialogmote-historikk og unntak/ikke-aktuell-historikk
+        DialogmotekandidatEndringArsak.DIALOGMOTE_FERDIGSTILT, DialogmotekandidatEndringArsak.DIALOGMOTE_LUKKET, DialogmotekandidatEndringArsak.UNNTAK, DialogmotekandidatEndringArsak.IKKE_AKTUELL -> null
```

Comment:
`DIALOGMOTE_FERDIGSTILT` og `DIALOGMOTE_LUKKET` trenger vi ikke ta med her siden de vil dekkes av historikken på dialogmøter. Eller tenker dere at de bør tas med i APIet for kompletthetens skyld?

Reply:
`UNNTAK` og `IKKE_AKTUELL` mappes under hvor vi også har data på veileder som gjorde vurderingen

Reply:
Blir kanskje litt på siden, men hvorfor vet `isdialogmotekandidat` om møtet er ferdigstilt og sånt? Det er jo `isdialogmote`-domene?

Reply:
Slik jeg leser koden er en arbeidstaker dialogmøtekandidat helt til et dialogmøte er ferdigstilt eller lukket (med mindre det settes unntak).

Reply:
Ah okey, ja det gir mening at endringene trigger noe. Tenker da at siden endringene egentlig hører hjemme i `isdialogmote` så trenger ikke APIet å returnere de her?

Reply:
Ja, stemmer det @geir-waagboe sier, det er derfor vi må vite om det her. Tror vi da er enige om at vi ikke tar med de i APIet :+1: